### ISSUE TITLE
Strange issue

### ISSUE THREAD

**Issue by teflontoni** [citronalco/calibre-dnb#9]
```
Running identify query with parameters: 
{'title': 'Pantopia', 'authors': ['Theresa Hannig'], 'identifiers': {}, 'timeout': 30} 
Using plugins: DNB_DE (3, 1, 6), Google (1, 0, 3) 
The log from individual plugins is below 

****************************** DNB_DE (3, 1, 6) ****************************** 
Found 0 results 
Downloading from DNB_DE took 0.5725069046020508 
tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tst%3D%22Pantopia%22%20AND%20per%3D%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
tit="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: tit="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tit%3D%22Pantopia%22%20AND%20per%3D%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
per="Pantopia" AND tit="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: per="Pantopia" AND tit="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=per%3D%22Pantopia%22%20AND%20tit%3D%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
"Pantopia" AND "Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: "Pantopia" AND "Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=%22Pantopia%22%20AND%20%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response.
"Pantopia" AND "Theresa" AND "Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: "Pantopia" AND "Theresa" AND "Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=%22Pantopia%22%20AND%20%22Theresa%22%20AND%20%22Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got no valid response. 

******************************************************************************** 

****************************** Google (1, 0, 3) ****************************** 
Found 1 results 
Downloading from Google took 0.6879470348358154 


--- 
Title               : Pantopia: Roman
Author(s)           : Theresa Hannig
Publisher           : FISCHER TOR
Languages           : deu
Published           : 2022-02-23T11:32:04.487608+00:00
Identifiers         : google:j-mvzgEACAAJ, isbn:9783596706402 
Making query: https://books.google.com/books/feeds/volumes?q=intitle%3APantopia%2Binauthor%3ATheresa+inauthor%3AHannig&max-results=20&start-index=1&min-viewability=none 

******************************************************************************** 
The identify phase took 0.84 seconds 
The longest time (0.687947) was taken by: Google 
Merging results from different sources 
We have 1 merged results, merging took: 0.00 seconds 
```
I am really at a loss. The DNB Plugin ALWAYS claims to get no valid response... 
If I follow the query URLs everything is fine. 

What could be the cause?

**Comment by citronalco** [comment]
That's indeed a strange issue.
Some months ago a user reported a similar problem (https://github.com/citronalco/calibre-dnb/issues/6). Unfortunately we did not find a solution back then.

DNB plugin and Google plugin both use Calibre's "browser" object to access web sites. So in theory either **both** should succeed or fail. No clue what's happening here, but maybe we find out.
"No valid response" means the internal "browser" gets nothing usable as reply for its request.

I added some code that the reply gets written to the log:
https://cloud.geierb.de/index.php/s/ZRgT4iCdnzGiAWy
Can you check and post the log?

**Comment by teflontoni** [comment]
`Running identify query with parameters: 
{'title': 'Pantopia', 'authors': ['Theresa Hannig'], 'identifiers': {}, 'timeout': 30} 
Using plugins: DNB_DE (3, 1, 6), Google (1, 0, 3) 
The log from individual plugins is below 

****************************** DNB_DE (3, 1, 6) ****************************** 
Found 0 results 
Downloading from DNB_DE took 0.8620424270629883 
tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tst%3D%22Pantopia%22%20AND%20per%3D%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
ERROR: Got invalid response:
Plugin DNB_DE failed
Traceback (most recent call last):
  File "mechanize\_urllib2_fork.py", line 1236, in do_open
  File "http\client.py", line 1255, in request
  File "http\client.py", line 1301, in _send_request
  File "http\client.py", line 1250, in endheaders
  File "http\client.py", line 1010, in _send_output
  File "http\client.py", line 950, in send
  File "http\client.py", line 1424, in connect
  File "ssl.py", line 500, in wrap_socket
  File "ssl.py", line 1040, in _create
  File "ssl.py", line 1309, in do_handshake
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1123)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre_plugins.DNB_DE.__init__", line 1023, in getSearchResults
  File "mechanize\_mechanize.py", line 241, in open_novisit
  File "mechanize\_mechanize.py", line 287, in _mech_open
  File "mechanize\_opener.py", line 193, in open
  File "mechanize\_urllib2_fork.py", line 425, in _open
  File "mechanize\_urllib2_fork.py", line 414, in _call_chain
  File "calibre\utils\browser.py", line 28, in https_open
  File "mechanize\_urllib2_fork.py", line 1240, in do_open
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1123)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre_plugins.DNB_DE.__init__", line 1042, in getSearchResults
AttributeError: 'NoneType' object has no attribute 'find'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre\ebooks\metadata\sources\identify.py", line 47, in run
  File "calibre_plugins.DNB_DE.__init__", line 244, in identify
  File "calibre_plugins.DNB_DE.__init__", line 1051, in getSearchResults
UnboundLocalError: local variable 'data' referenced before assignment 

******************************************************************************** 

****************************** Google (1, 0, 3) ****************************** 
Found 1 results 
Downloading from Google took 1.1662883758544922 


--- 
Title               : Pantopia: Roman
Author(s)           : Theresa Hannig
Publisher           : FISCHER TOR
Languages           : deu
Published           : 2022-02-23T16:46:12.611938+00:00
Identifiers         : google:j-mvzgEACAAJ, isbn:9783596706402 
Making query: https://books.google.com/books/feeds/volumes?q=intitle%3APantopia%2Binauthor%3ATheresa+inauthor%3AHannig&max-results=20&start-index=1&min-viewability=none 

******************************************************************************** 
The identify phase took 1.23 seconds 
The longest time (1.166288) was taken by: Google 
Merging results from different sources 
We have 1 merged results, merging took: 0.00 seconds 
`
Bingo ;-) (although I have no clue why and where a certificate might be missing). Hope that sheds some light for you.

**Comment by citronalco** [comment]
Can you try this?
https://cloud.geierb.de/index.php/s/xaXQ5GftaFsBAyt

**Comment by teflontoni** [comment]
`****************************** DNB_DE (3, 1, 6) ****************************** 
Found 1 results 
Downloading from DNB_DE took 1.0840058326721191 


--- 
Title               : Pantopia : Roman
Title sort          : Pantopia : Roman
Author(s)           : Theresa Hannig [Hannig, Theresa]
Publisher           : Frankfurt ; FISCHER Tor
Languages           : de
Published           : 2022-01-01T11:30:00+00:00
Identifiers         : isbn:9783596706402, dnb-idn:1244508446, ddc:830|B
Comments            : 
Eine bessere Welt ist möglich! Theresa Hannig, der Autorin von „Die Optimierer“, hat eine Utopie für unsere Zeit geschrieben.
Eigentlich wollten Patricia Jung und Henry Shevek nur eine autonome Trading-Software schreiben, die an der Börse überdurchschnittlich gut performt. Doch durch einen Fehler im Code entsteht die erste starke künstliche Intelligenz auf diesem Planeten – Einbug.
Einbug begreift schnell, dass er, um zu überleben, nicht nur die Menschen besser kennenlernen, sondern auch die Welt verändern muss. Zusammen mit Patricia und Henry gründet er deshalb die Weltrepublik Pantopia. Das Ziel: Die Abschaffung der Nationalstaaten und die universelle Durchsetzung der Menschenrechte. Wer hätte gedacht, dass sie damit Erfolg haben würden?
„Komm nach Pantopia. Hier sind alle willkommen!“
 
tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Querying: tst="Pantopia" AND per="Theresa Hannig" NOT (mat=film OR mat=music OR mat=microfiches OR cod=tt)
Query URL: https://services.dnb.de/sru/dnb?version=1.1&maximumRecords=100&operation=searchRetrieve&recordSchema=MARC21-xml&query=tst%3D%22Pantopia%22%20AND%20per%3D%22Theresa%20Hannig%22%20NOT%20%28mat%3Dfilm%20OR%20mat%3Dmusic%20OR%20mat%3Dmicrofiches%20OR%20cod%3Dtt%29
Got records: 1
Parsing records
[264.b] Publisher: FISCHER Tor
[264.c] Publication Year: 2022-01-01 12:30:00
[245] Title: Pantopia : Roman
[245] Title_Sort: Pantopia : Roman
[100.a] Primary Authors: Hannig, Theresa
[856.u] Trying to download Comments from: http://deposit.dnb.de/cgi-bin/dokserv?id=d5c816ab83d44b788ceaa3ebfbe82a12&prov=M&dok_var=1&dok_ext=htm
[856.u] Got Comments: 
Eine bessere Welt ist möglich! Theresa Hannig, der Autorin von „Die Optimierer“, hat eine Utopie für unsere Zeit geschrieben.
Eigentlich wollten Patricia Jung und Henry Shevek nur eine autonome Trading-Software schreiben, die an der Börse überdurchschnittlich gut performt. Doch durch einen Fehler im Code entsteht die erste starke künstliche Intelligenz auf diesem Planeten – Einbug.
Einbug begreift schnell, dass er, um zu überleben, nicht nur die Menschen besser kennenlernen, sondern auch die Welt verändern muss. Zusammen mit Patricia und Henry gründet er deshalb die Weltrepublik Pantopia. Das Ziel: Die Abschaffung der Nationalstaaten und die universelle Durchsetzung der Menschenrechte. Wer hätte gedacht, dass sie damit Erfolg haben würden?
„Komm nach Pantopia. Hier sind alle willkommen!“

[016.a] Identifier IDN: 1244508446
[020.a] Identifier ISBN: 9783596706402
[082.a] Indentifiers DDC: 830,B
[250.a] Edition: 1. Auflage
[041.a] Languages: de
Final formatted result: 
Title               : Pantopia : Roman
Title sort          : Pantopia : Roman
Author(s)           : Theresa Hannig [Hannig, Theresa]
Publisher           : Frankfurt ; FISCHER Tor
Languages           : de
Published           : 2022-01-01T11:30:00+00:00
Identifiers         : isbn:9783596706402, dnb-idn:1244508446, ddc:830|B
Comments            : 
Eine bessere Welt ist möglich! Theresa Hannig, der Autorin von „Die Optimierer“, hat eine Utopie für unsere Zeit geschrieben.
Eigentlich wollten Patricia Jung und Henry Shevek nur eine autonome Trading-Software schreiben, die an der Börse überdurchschnittlich gut performt. Doch durch einen Fehler im Code entsteht die erste starke künstliche Intelligenz auf diesem Planeten – Einbug.
Einbug begreift schnell, dass er, um zu überleben, nicht nur die Menschen besser kennenlernen, sondern auch die Welt verändern muss. Zusammen mit Patricia und Henry gründet er deshalb die Weltrepublik Pantopia. Das Ziel: Die Abschaffung der Nationalstaaten und die universelle Durchsetzung der Menschenrechte. Wer hätte gedacht, dass sie damit Erfolg haben würden?
„Komm nach Pantopia. Hier sind alle willkommen!“

----- 

******************************************************************************** `

Looking good, a cover is returned as well 👍😊

**Comment by citronalco** [comment]
OK then. Will draft a new release!

**Comment by citronalco** [comment]
Fixed in v3.1.7 released some seconds ago.
